docs: Clarify the use of Gutenberg and Gutenberg Mobile repository names

derekblank opened the pull request:
Clarifies the references to Gutenberg and Gutenberg Mobile repositories in E2E testing docs.

@dcalhoun in the context of previous changes in 4574af6bad0731ef4e19eaddf74c5da0ba8e6ead, please verify if the intent in this PR is correct, and feel free to make and changes or suggestions directly. 


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

derekblank committed:
docs: Clarify the use of Gutenberg and Gutenberg Mobile repository names

See also:
  4574af6bad0731ef4e19eaddf74c5da0ba8e6ead

fluiddot commented on README.md:
```diff
@@ -125,11 +125,11 @@ This project is set up to use [jest](https://facebook.github.io/jest/) for tests
 ## End-to-End Tests
 
-This repository extends the end-to-end (E2E) tests found in the Mobile Gutenberg (MG) [project repository](https://github.com/WordPress/gutenberg). The majority of E2E tests should reside in the MG repository. However, experimental tests or non-core-related tests should be placed in the `gutenberg-mobile` repository.
+This repository extends the end-to-end (E2E) tests found in the [Gutenberg project repository](https://github.com/WordPress/gutenberg). The majority of E2E tests should reside in the `gutenberg` repository. However, experimental tests or non-core-related tests should be placed in the `gutenberg-mobile` repository.
```
> The majority of E2E tests should reside in the `gutenberg` repository. However, experimental tests or non-core-related tests should be placed in the `gutenberg-mobile` repository.

This is true as ideally, all E2E tests should reside in Gutenberg. However, the reality is that we only run a basic E2E test ([reference](https://github.com/WordPress/gutenberg/blob/6ef4629f77ab3da6502508b4c9539a3912300f0e/.github/workflows/rnmobile-ios-runner.yml#L23)) due to limitations. Eventually, we'd need to explore how to migrate E2E tests to be run in Gutenberg, but in the meantime I'd propose clarifying here the situation.

dcalhoun replied:
I think the suggestion to clearly communicate the current reality that we do not place most E2E tests in `gutenberg` makes sense. I'll update the PR to address this.

fluiddot commented on README.md:
```diff
@@ -125,11 +125,11 @@ This project is set up to use [jest](https://facebook.github.io/jest/) for tests
+This repository extends the end-to-end (E2E) tests found in the [Gutenberg project repository](https://github.com/WordPress/gutenberg). The majority of E2E tests should reside in the `gutenberg` repository. However, experimental tests or non-core-related tests should be placed in the `gutenberg-mobile` repository.
 
-Reviewing the MG E2E test [documentation](https://github.com/WordPress/gutenberg/tree/trunk/packages/react-native-editor/__device-tests__#readme) is the best approach for understanding the test environment and how to set up your computer to run tests. Much of the information and approaches outlined there also apply to the tests found in this repository. The only difference being that this reposit ...
+Reviewing the [Mobile Gutenberg E2E test documentation](https://github.com/WordPress/gutenberg/tree/trunk/packages/react-native-editor/__device-tests__#readme) is the best approach for understanding the test environment and how to set up your computer to run tests. Much of the information and approaches outlined there also apply to the tests found in this repository. The only difference being tha ...
```
Not a blocker, but I noticed we reference the project differently across this file. The title says `Mobile Gutenberg` and in other areas ([example](https://github.com/wordpress-mobile/gutenberg-mobile/blob/4586443f1b2bbc3483923b457c637208a269afe7/README.md#internationalization-i18n)) it says `Gutenberg Mobile`. It would be great if we use the same name for consistency.

dcalhoun replied:
I agree there is opportunity to improve consistency and clarity. 

For this line in particular, this sentence intended to reference `gutenberg`, not `gutenberg-mobile`. So, `MG` should likely be replaced with `gutenberg` to match earlier proposed changes on line ~128. I'll update the PR to address this.

dcalhoun commented on README.md:
```diff
@@ -125,11 +125,11 @@ This project is set up to use [jest](https://facebook.github.io/jest/) for tests
+Reviewing the [Mobile Gutenberg E2E test documentation](https://github.com/WordPress/gutenberg/tree/trunk/packages/react-native-editor/__device-tests__#readme) is the best approach for understanding the test environment and how to set up your computer to run tests. Much of the information and approaches outlined there also apply to the tests found in this repository. The only difference being tha ...
 
-After reviewing the MG E2E test documentation, the following examples showcase running the E2E tests found in this repository using the repositories scripts.
+After reviewing the Mobile Gutenberg E2E test documentation, the following examples showcase running the E2E tests found in this repository using the repositories scripts.
```
For this line in particular, this sentence intended to reference `gutenberg`, not `gutenberg-mobile`. So, `MG` should likely be replaced with `gutenberg` to match earlier proposed changes on line ~128. I'll update the PR to address this.

dcalhoun reviewed (commented):
Thank you for proposing changes to improve clarity. I will follow up and move this forward.

dcalhoun committed:
docs: Improve project name consistency

Attempt to mitigate confusion between the `gutenberg` and
`gutenberg-mobile` repositories, in addition to the general "Gutenberg"
project and plugin.

dcalhoun committed:
docs: Clarify the current reality of E2E test locations

It is better to describe the current state, rather than an ideal.

dcalhoun commented:
I updated the PR to rely upon code repository names to be as explicit as possible during this period where we do not necessarily have a clear and consistently used project name to described the mobile variant of the block editor and differences between the two repositories.

fluiddot reviewed (approved):
LGTM 🎊 ! Thanks @dcalhoun for updating the documentation 🙇 !

I noticed that some of the CI checks are failing. The error is the same we encountered yesterday related to Buildkite not being able to checkout the Gutenberg repository (https://github.com/wordpress-mobile/gutenberg-mobile/pull/6520#issuecomment-1882916737). This is already fixed in `trunk` (https://github.com/wordpress-mobile/gutenberg-mobile/pull/6525), so I'd recommend updating the branch with the latest changes.

dcalhoun committed:
Merge branch 'trunk' of github.com:wordpress-mobile/gutenberg-mobile into docs/clarify-e2e-test-repo-names
